### PR TITLE
Add test coverage for the AddonRecommender

### PR DESCRIPTION
**Pull request by Dexterp37**
This PR adds _basic_ test coverage for the `AMODataset` and the `AddonRecommender` scala objects. Long due, I know

### CODE REVIEW

**Comment by Dexterp37** [comment]
Rotfl, what about I actually add the test files to the commit? :-)

File: src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala [hunk 1]
```diff
@@ -94,6 +93,37 @@ object AddonRecommender {
...
   }
 
+  /**
+    * Get a new dataset from the Longitudinal dataset containing only the relevant addon data.
+    * @param sqlContext The sql context for loading the Longitudinal dataset.
+    * @param addonBlacklist The a list of addon GUIDs to exclude.
+    * @return A 4 columns Dataset with each row having the client id, the addon GUID
+    *         and the hashed client id and GUID.
+    */
+  private def getAddonData(sqlContext: SQLContext, addonBlacklist: List[String]) = {
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala]: Pass in the `SparkSession` here instead (you can use it just the same - `spark.sql("SELECT * ...")`).

File: src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala [hunk 2]
```diff
@@ -129,33 +159,14 @@ object AddonRecommender {
...
 
   private def train(localOutputDir: String, runDate: String, privateBucket: String, publicBucket: String) = {
-    val sparkConf = new SparkConf().setAppName(this.getClass.getName)
-    sparkConf.setMaster(sparkConf.get("spark.master", "local[*]"))
-    val sc = new SparkContext(sparkConf)
-    val hiveContext = new HiveContext(sc)
-    import hiveContext.implicits._
+    val spark = getOrCreateSparkSession("AddonRecommenderTest", enableHiveSupport = true)
+    val sc = spark.sparkContext
+    import spark.implicits._
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala]: Is this line required now, since it is also above?

File: src/main/scala/com/mozilla/telemetry/views/SyncFlatView.scala [hunk 3]
```diff
@@ -76,7 +76,7 @@ object SyncFlatView {
         case "sync" => true
       }.where("submissionDate") {
         case date if date == currentDate.toString("yyyyMMdd") => true
-      }.records(conf.limit.get, Some(partitionCount))
+      }.records(conf.limit.get)
```
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/views/SyncFlatView.scala]: This should be a separate commit, no?

File: src/test/scala/com/mozilla/telemetry/ml/AMODatabaseTest.scala [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration._
+import org.json4s.MappingException
+import org.scalatest.{BeforeAndAfterEach, FlatSpec, Matchers}
+
+class AMODatabaseTest extends FlatSpec with Matchers with BeforeAndAfterEach {
+  val Port = 3785
+  val Host = "localhost"
```
Comment (fbertsch) [inline src/test/scala/com/mozilla/telemetry/ml/AMODatabaseTest.scala]: I really really wish there was a way to completely stub a remote resource without using `localhost` :/

File: src/main/scala/com/mozilla/telemetry/views/SyncFlatView.scala (hunk 3 above)
Comment (Dexterp37) [reply src/main/scala/com/mozilla/telemetry/views/SyncFlatView.scala]: Whoops, that was a mistake, I didn't mean to commit that.

File: src/test/scala/com/mozilla/telemetry/ml/AMODatabaseTest.scala (hunk 4 above)
Comment (Dexterp37) [reply src/test/scala/com/mozilla/telemetry/ml/AMODatabaseTest.scala]: Yeah, me too :(

File: src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala (hunk 2 above)
Comment (Dexterp37) [reply src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala]: I'm afraid it is: without this line, the one slightly below `.map{ case (_, _, hashedClientId, hashedAddonId) => Rating(hashedClientId, hashedAddonId, 1.0f)}` complains about not finding the encoder and suggests to import the implicits.

File: src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala (hunk 1 above)
Comment (Dexterp37) [reply src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala]: Done!
